### ISSUE TITLE
Loss Function

### ISSUE THREAD

**Issue by wieschoo** [ikostrikov/TensorFlow-VAE-GAN-DRAW#4]
Hi,

good work @ikostrikov ! I just wonder where the loss function:
https://github.com/ikostrikov/TensorFlow-VAE-GAN-DRAW/blob/master/main-gan.py#L60-L61
comes from?
Because it does not look like a cross entropy loss
![cross](https://cloud.githubusercontent.com/assets/3343426/13262116/90861522-da63-11e5-8482-0a10b8e8b202.png)

I hope you do not mind my way of asking here.

**Comment by ikostrikov** [comment]
Hi!

Thanks!

D1 and D2 have fixed labels there there was no need to write them down explicitly. Then, I took a more numerically stable formulation from:
https://www.tensorflow.org/versions/r0.7/api_docs/python/nn.html#sigmoid_cross_entropy_with_logits

Yes, sure. No problem.
